clusterloader2/docs/network-perf-testcase-enhancement/README.md

```diff
@@ -0,0 +1,582 @@
+   are run on same worker node, then more then 2 vCPUs might be required.
+
+### Future Enhancements
+
+Performance benchmarking frameworks can be enhanced to include testcases related to various CNIs.
+
+### Benefits
+
+1) Our proposal attempts to standardize the test scenarios and cover most common performance metrics.
+2) Detecting performance degradation in early phase of feature delivery.
+3) With standard test scenarios available, kubernetes users will be able adopt these test frameworks
+   to validate their downstream offerings.  
+   
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+Benchmarking and measurements with intermediate services like ingress controllers, load-balancers 
+are out of scope of this proposal.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+Kubernetes uses the Prow framework for CI/CD and the new network performance related tests are also planned to be added as part of prow jobs.
+
+There were multiple test frameworks analysed for running the test cases like: k8s E2E tests and sig-scalability’s Netperf and ClusterLoader2 frameworks.
+Since we are planning to have the test cases under sig scalability where ClusterLoader2 is already running many test cases as part of kubernetes CI/CD and since the framework is easy to extend for creating new tests and test scenarios ,it is decided to use ClusterLoader2 for the implementation.These tests needs a statistical distribution on a large number of pods to smooth out infrastructure dependencies  and cluster loader framework already run tests with similar requirements.
+
+![Image of test run](hld-pic1.PNG)
+
+
+The below 3PP tools will be used for benchmarking network performance:
+* Iperf2
+* Iperf3
+* Siege
+
+These tools would be included as part of a container image and would be run as pods(as client pods and server pods) for collecting measurements.
+
+There are 2 design options identified to implement these test cases in ClusterLoader:
+* Measurement using Probes
+* Phases
+
+### Measurement using Probes (Option1)
```

Comment:
I've been thinking a bit about it, and I think we should proceed with this option, with the following difference:
- let's NOT create a controller
- instead, just implement the "pods" that will be running in the cluster and use the new measurement you will introduce as a "controller"